eos-paygd/main.c

```diff
@@ -532,7 +532,6 @@ main (int   argc,
       else if (g_error_matches (error, GSS_SERVICE_ERROR, GSS_SERVICE_ERROR_SIGNALLED))
         {
           g_warning ("EpgService exited with GSS_SERVICE_ERROR_SIGNALLED");
-          raise (gss_service_get_exit_signal (GSS_SERVICE (service)));
```

Comment:
This dates all the way back to the initial commit in 150a34491830d49ec1a766c1eb617389effb372e. @pwithnall do you recall the rationale here?

Reply:
`GSS_SERVICE_ERROR_SIGNALLED` is returned if a signal (by default, `SIGINT` or `SIGTERM`) is caught by the service. Rather than have the signal handlers try and clean up and exit, it seemed cleaner to just use that as a trigger for exiting the `gss_service_run()` call (which runs the main loop), and have the caller (i.e. this code) work out what it wants to do with the signal. Re-`raise()`ing it seemed like the correct thing to do in the absence of other requirements.

The code may want to do some process-wide cleanup or state saving before re-`raise()`ing the signal.

Essentially, the change proposed here means that the process will now swallow `SIGINT` and `SIGTERM` and, rather than exiting via `raise()`, it will warn and return `FATAL_SIGNAL_EXIT_CODE` from `main()`.

Reply:
Changed my mind slightly here while addressing other points. It does make sense to re-raise the signal for exiting so systemd has an accurate picture of what happened, but just at a different place now that we want to request a poweroff when that happen.

Reply:
:+1: :+1: